composer/workflows/dataproc_batch_operators.py

```diff
@@ -0,0 +1,169 @@
+# for e.g. projects/my-project/locations/my-region/services/my-cluster
+CUSTOM_CONTAINER = "us.gcr.io/{{var.value.project_id}}/{{ var.value.image_name}}"
+# for e.g. "us.gcr.io/my-project/quickstart-image",
+
+default_args = {
+    # Tell airflow to start one day ago, so that it runs as soon as you upload it
+    "start_date": days_ago(1),
+    "project_id": PROJECT_ID,
+    "region": REGION,
+
+}
+with models.DAG(
+    "dataproc_batch_operators",  # The id you will see in the DAG airflow page
+    default_args=default_args,  # The interval with which to schedule the DAG
+    schedule_interval=datetime.timedelta(days=1),  # Override to match your needs
+) as dag:
+
+    create_batch = DataprocCreateBatchOperator(
+        task_id="batch_create",
+        batch={
+            "pyspark_batch": {
+                "main_python_file_uri": PYTHON_FILE_LOCATION,
+                "jar_file_uris": [SPARK_BIGQUERY_JAR_FILE],
+            },
+            "environment_config": {
+                "peripherals_config": {
+                    "spark_history_server_config": {
+                        "dataproc_cluster": PHS_CLUSTER_PATH,
+                    },
+                },
+            },
+        },
+        batch_id="batch-create-phs",
+    )
+    # [START composer_dataproc_list_batch]
+    list_batches = DataprocListBatchesOperator(
+        task_id="list-all-batches",
+    )
+    # [END composer_dataproc_list_batch]
+
+    # [START composer_dataproc_get_batch]
+    get_batch = DataprocGetBatchOperator(
+        task_id="get_batch",
+        batch_id="batch-create-phs",
+    )
+    # [END composer_dataproc_get_batch]
+    # [START composer_dataproc_delete_batch]
+    delete_batch = DataprocDeleteBatchOperator(
+        task_id="delete_batch",
+        batch_id="batch-create-phs",
+    )
+    # [END composer_dataproc_delete_batch]
+    create_batch >> list_batches >> get_batch >> delete_batch
+    # [END composer_dataproc_create_batch]
+
+    # [START composer_dataproc_create_metastore_batch]
+    create_batch_with_metastore = DataprocCreateBatchOperator(
+        task_id="dataproc_metastore",
+        batch={
+            "pyspark_batch": {
+                "main_python_file_uri": PYTHON_FILE_LOCATION,
+                "jar_file_uris": [SPARK_BIGQUERY_JAR_FILE],
+            },
+            "environment_config": {
+                "peripherals_config": {
+                    "metastore_service": METASTORE_SERVICE_LOCATION,
+                    "spark_history_server_config": {
+                        "dataproc_cluster": PHS_CLUSTER_PATH,
+                    },
+                 },
+            },
+        },
+        batch_id="dataproc-metastore",
+    )
+    get_batch_metastore = DataprocGetBatchOperator(
+        task_id="get_batch_metatstore",
+        batch_id="dataproc-metastore",
+    )
+    delete_batch_metastore = DataprocDeleteBatchOperator(
+        task_id="delete_batch_metastore",
+        batch_id="dataproc-metastore",
+    )
```

Comment:
Do we need to show the get / delete operators for each example as opposed to just showing it once in this doc?

Reply:
Do you mean having them in the DAG at all or having them show up in the doc with clever region tag usage? I think that they should be in the DAG just so that the DAG shows complete crud operations for each one and also doesn't leave users with hanging resources after running the DAG.

Reply:
Agreed. Keeping it :) for a clean dataproc batches list at the end.